lib/plugin/instance.rb

```diff
@@ -516,6 +516,23 @@ def after_initialize(&block)
     initializers << block
   end
 
+  def commit_hash(short: false)
+    command = "git rev-parse HEAD".split(" ")
+    command = command.insert(2, "--short") if short
+    Discourse::Utils.execute_command(*command, chdir: directory).strip
+  rescue Discourse::Utils::CommandError => e
+    Rails.logger.warn("Unable to get commit hash for plugin #{name}: #{e}")
+    ""
+  end
```

Comment:
I think we can improve this further by eliminating the call to this at runtime and for each unicorn worker process by preloading this. See https://github.com/discourse/discourse/blob/e2e17bfb0f56092ccb8167b5fc39983538959d30/lib/discourse.rb#L1079